seqset.go

```diff
@@ -317,3 +317,26 @@ func (s SeqSet) search(q uint32) (i int, ok bool) {
 	}
 	return min, s[min].Contains(q)
 }
+
+func intersectSeqSet(a, b SeqSet) SeqSet {
+	// TODO: optimize
+	var out SeqSet
+	for _, seq := range a {
+		for _, other := range b {
+			// TODO: handle '*'
```

Comment:
This is actually tricky to handle. For instance:

```
42 EXISTS
SEARCH 1:* 1:2
-> 1:2
SEARCH 1:* 1:60
-> 1:*
SEARCH * 20
-> nil
SEARCH * 42
-> *
```

I think we need to either accumulate multiple seqsets per criteria, either track the number of messages in the selected mailbox in the server connection logic.

Reply:
Removed the need for this in #523